src/core/playground/authenticationKey.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { sha3_256 as sha3Hash } from "@noble/hashes/sha3";
+import { HexInput } from "../../types";
+
+import { AccountAddress } from "../accountAddress";
+import { Hex } from "../hex";
+
+/**
+ * Each account stores an authentication key. Authentication key enables account owners to rotate
+ * their private key(s) associated with the account without changing the address that hosts their account.
+ * @see {@link https://aptos.dev/concepts/accounts | Account Basics}
+ *
+ * Note: AuthenticationKey only supports Ed25519 and MultiEd25519 public keys for now.
+ *
+ * Account addresses can be derived from AuthenticationKey
+ */
+export class AuthenticationKey {
+  /**
+   * An authentication key is always a SHA3-256 hash of data, and is always 32 bytes.
+   */
+  static readonly LENGTH: number = 32;
+
+  /**
+   * The raw bytes of the authentication key.
+   */
+  public readonly data: Hex;
+
+  constructor(args: { data: HexInput }) {
+    const { data } = args;
+    const hex = Hex.fromHexInput(data);
+    if (hex.toUint8Array().length !== AuthenticationKey.LENGTH) {
+      throw new Error(`Authentication Key length should be ${AuthenticationKey.LENGTH}`);
+    }
+    this.data = hex;
+  }
+
+  toString(): string {
+    return this.data.toString();
+  }
+
+  toUint8Array(): Uint8Array {
+    return this.data.toUint8Array();
+  }
+
+  // Extracted the common logic for authkey derivation and put it here.
+  // Don't love the name but we can find something better
+  static fromHashInput({ hashInput }: { hashInput: HexInput }): AuthenticationKey {
```

Comment:
I think we can use here one input param
```
static fromHashInput(authKeyBytes):
```

Reply:
This should... be instead fromSchemeAndBytes({scheme: Scheme, input: HexInput)

It should take the scheme, and the bytes required to generate the authentication key.

Right now, it's unclear what I'd put here since, if it's the public key, I still need to prepend (or append I forget), with the scheme to get the correct auth key.

The input, may be a key, it may be a set of keys etc.

Reply:
> This should... be instead fromSchemeAndBytes({scheme: Scheme, input: HexInput)

I like it